dcos/client.go

```diff
@@ -38,9 +53,100 @@ func NewClientWithOptions(httpClient *http.Client, config *Config) (*Client, err
+	dcosURL, err := url.Parse(clusterURL)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid DC/OS cluster URL '%s': %v", clusterURL, err)
+	}
+
+	iamRuntime := runtimeClient.NewWithClient(
+		dcosURL.Host,
+		iamclient.DefaultBasePath,
+		[]string{dcosURL.Scheme},
+		client,
+	)
+
+	return iamclient.New(iamRuntime, nil), nil
+}
+
+func newSecretsClient(clusterURL string, client *http.Client) (*secretsclient.DCOSSecrets, error) {
+	dcosURL, err := url.Parse(clusterURL)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid DC/OS cluster URL '%s': %v", clusterURL, err)
+	}
+
+	secretsRuntime := runtimeClient.NewWithClient(
+		dcosURL.Host,
+		secretsclient.DefaultBasePath,
+		[]string{dcosURL.Scheme},
+		client,
+	)
+
+	return secretsclient.New(secretsRuntime, nil), nil
+}
+
+func newCosmosClient(clusterURL string, client *http.Client) (*cosmosclient.Cosmos, error) {
+	dcosURL, err := url.Parse(clusterURL)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid DC/OS cluster URL '%s': %v", clusterURL, err)
+	}
+
+	cosmosRuntime := runtimeClient.NewWithClient(
+		dcosURL.Host,
+		cosmosclient.DefaultBasePath,
+		[]string{dcosURL.Scheme},
+		client,
+	)
+
+	cosmosRuntime.Consumers["application/vnd.dcos.package.install-response+json"] = cosmosRuntime.Consumers[runtime.JSONMime]
+	cosmosRuntime.Producers["application/vnd.dcos.package.install-request+json;charset=utf-8;version=v1"] = cosmosRuntime.Producers[runtime.JSONMime]
+	cosmosRuntime.Consumers["application/vnd.dcos.package.uninstall-response+json"] = cosmosRuntime.Consumers[runtime.JSONMime]
+	cosmosRuntime.Producers["application/vnd.dcos.package.uninstall-request+json;charset=utf-8;version=v1"] = cosmosRuntime.Producers[runtime.JSONMime]
+	cosmosRuntime.Consumers["application/vnd.dcos.package.error+json"] = cosmosRuntime.Consumers[runtime.JSONMime]
+	cosmosRuntime.Consumers["application/vnd.dcos.service.update-response+json"] = cosmosRuntime.Consumers[runtime.JSONMime]
+	cosmosRuntime.Producers["application/vnd.dcos.service.update-request+json;charset=utf-8;version=v1"] = cosmosRuntime.Producers[runtime.JSONMime]
+	cosmosRuntime.Consumers["application/vnd.dcos.package.repository.add-response+json"] = cosmosRuntime.Consumers[runtime.JSONMime]
+	cosmosRuntime.Producers["application/vnd.dcos.package.repository.add-request+json;charset=utf-8;version=v1"] = cosmosRuntime.Producers[runtime.JSONMime]
+	cosmosRuntime.Consumers["application/vnd.dcos.package.repository.delete-response+json"] = cosmosRuntime.Consumers[runtime.JSONMime]
+	cosmosRuntime.Producers["application/vnd.dcos.package.repository.delete-request+json;charset=utf-8;version=v1"] = cosmosRuntime.Producers[runtime.JSONMime]
+	cosmosRuntime.Consumers["application/vnd.dcos.service.describe-response+json"] = cosmosRuntime.Consumers[runtime.JSONMime]
+	cosmosRuntime.Producers["application/vnd.dcos.service.describe-request+json;charset=utf-8;version=v1"] = cosmosRuntime.Producers[runtime.JSONMime]
```

Comment:
Can these be part of the spec and autogenerated?

Reply:
That would be nice, but I only found this: https://github.com/go-openapi/runtime/issues/33